oc-chef-pedant/spec/api/oc_id_spec.rb

```diff
@@ -17,6 +18,47 @@
     end
   end
 
+  context "key reset" do
+    let(:username) { platform.non_admin_user.name }
+    let(:password) { "foobar" } #hardcoded at the platform layer
+
+    let(:request_headers) do
+      {
+        "Content-Type" => "application/x-www-form-urlencoded",
+        "Cookie" => csrf[:cookie]
+      }
+    end
+
+    let(:csrf) do
+      response = get("#{platform.server}/id/signin", platform.superuser, headers: {"Accept" => "text/html"})
+      cookie = response.headers[:set_cookie][1].split(";").first
+      # I KNOW. I'll leave it up to reviewers whether we should pull
+      # in nokogiri or hpricot just do to this
```

Comment:
I'm alright with leaving this. Maybe when we expand these tests further we can look at either making this part of the testing lib, or changing to nokigiri.